dataset/src/main/scala/frameless/TypedDataset.scala

```diff
@@ -1036,8 +1036,15 @@ object TypedDataset {
     val shouldReshape = output.zip(targetColNames).exists {
       case (expr, colName) => expr.name != colName
     }
-
-    val reshaped = if (shouldReshape) df.toDF(targetColNames: _*) else df
+    val canSelect = targetColNames.toSet.subsetOf(output.map(_.name).toSet)
```

Comment:
Can you give an example where this would be false?

Reply:
If you'd remove the `.toDF("b", "a")` call from the test case this one would be such a case where the column names are still `_1` and `_2` from the original dataframe.
Frameless would be successful here if the types would align.